json_serialization_writer.go

```diff
@@ -585,350 +585,232 @@ func (w *JsonSerializationWriter) GetSerializedContent() ([]byte, error) {
-				err := w.WriteTimeValue(key, &tv)
-				if err != nil {
-					return err
-				}
-				continue
+		}
+	}
+	return nil
+}
+
+// writeAdditional Helper function for writing additional data. Feature adds
+// the ability to support certain map types.
+func (w *JsonSerializationWriter) writeAdditional(key string, input any) error {
+	var err error
+	switch value := input.(type) {
+	case absser.Parsable:
+		err = w.WriteObjectValue(key, value)
+	case []absser.Parsable:
+		err = w.WriteCollectionOfObjectValues(key, value)
+	case []string:
+		err = w.WriteCollectionOfStringValues(key, value)
+	case []bool:
+		err = w.WriteCollectionOfBoolValues(key, value)
+	case []byte:
+		err = w.WriteCollectionOfByteValues(key, value)
+	case []int8:
+		err = w.WriteCollectionOfInt8Values(key, value)
+	case []int32:
+		err = w.WriteCollectionOfInt32Values(key, value)
+	case []int64:
+		err = w.WriteCollectionOfInt64Values(key, value)
+	case []float32:
+		err = w.WriteCollectionOfFloat32Values(key, value)
+	case []float64:
+		err = w.WriteCollectionOfFloat64Values(key, value)
+	case []uuid.UUID:
+		err = w.WriteCollectionOfUUIDValues(key, value)
+	case []time.Time:
+		err = w.WriteCollectionOfTimeValues(key, value)
+	case []absser.ISODuration:
+		err = w.WriteCollectionOfISODurationValues(key, value)
+	case []absser.TimeOnly:
+		err = w.WriteCollectionOfTimeOnlyValues(key, value)
+	case []absser.DateOnly:
+		err = w.WriteCollectionOfDateOnlyValues(key, value)
+	case *string:
+		err = w.WriteStringValue(key, value)
+	case string:
+		err = w.WriteStringValue(key, &value)
+	case *bool:
+		err = w.WriteBoolValue(key, value)
+	case bool:
+		err = w.WriteBoolValue(key, &value)
+	case *byte:
+		err = w.WriteByteValue(key, value)
+	case byte:
+		err = w.WriteByteValue(key, &value)
+	case *int8:
+		err = w.WriteInt8Value(key, value)
+	case int8:
+		err = w.WriteInt8Value(key, &value)
+	case *int32:
+		err = w.WriteInt32Value(key, value)
+	case int32:
+		err = w.WriteInt32Value(key, &value)
+	case *int64:
+		err = w.WriteInt64Value(key, value)
+	case int64:
+		err = w.WriteInt64Value(key, &value)
+	case *float32:
+		err = w.WriteFloat32Value(key, value)
+	case float32:
+		err = w.WriteFloat32Value(key, &value)
+	case *float64:
+		err = w.WriteFloat64Value(key, value)
+	case float64:
+		err = w.WriteFloat64Value(key, &value)
+	case *uuid.UUID:
+		err = w.WriteUUIDValue(key, value)
+	case uuid.UUID:
+		err = w.WriteUUIDValue(key, &value)
+	case *time.Time:
+		err = w.WriteTimeValue(key, value)
+	case time.Time:
+		err = w.WriteTimeValue(key, &value)
+	case *absser.ISODuration:
+		err = w.WriteISODurationValue(key, value)
+	case absser.ISODuration:
+		err = w.WriteISODurationValue(key, &value)
+	case *absser.TimeOnly:
+		err = w.WriteTimeOnlyValue(key, value)
+	case absser.TimeOnly:
+		err = w.WriteTimeOnlyValue(key, &value)
+	case *absser.DateOnly:
+		err = w.WriteDateOnlyValue(key, value)
+	case absser.DateOnly:
+		err = w.WriteDateOnlyValue(key, &value)
+	case *JsonParseNode:
```

Comment:
This was a generic way to add the capability needed for `dateTime` and `timeZone` functionality. A more robust handler for these types could be created and tested if the parsing of eventMessageRequest is going to remain the same.

Reply:
Decided that this would not be implemented in this branch. See Issue #30